src/data-access/GroupRepository.js

```diff
@@ -1,76 +1,52 @@
-        if (groupToUpdate) {
-            const result = await database.Group.update(group, { where: { id } });
-            if (result > 0) {
-                return group;
-            }
+    const groupToUpdate = await database.Group.findOne({ where: { id } });
+    if (groupToUpdate) {
+        const result = await database.Group.update(group, { where: { id } });
+        if (result > 0) {
+            return group;
         }
-        return null;
-    } catch (error) {
-        throw error;
     }
+    return null;
 };
 
 export const removeGroup = async (id) => {
-    try {
-        const toRemove = await database.Group.findOne({ where: { id } });
-        if (toRemove) {
-            const result = await database.Group.destroy({ where: { id } });
-            if (result > 0) {
-                return toRemove;
-            }
+    const toRemove = await database.Group.findOne({ where: { id } });
+    if (toRemove) {
+        const result = await database.Group.destroy({ where: { id } });
+        if (result > 0) {
+            return toRemove;
         }
-        return null;
-    } catch (error) {
-        throw error;
     }
+    return null;
 };
 
 export const saveUserGroup = async (groupId, userIds) => {
-    try {
-        const group = await database.Group.findOne({ where: { id : groupId } });
-        if (group) {
-            const users = await database.User.findAll({ where: { id : userIds, isDeleted : false } });
-            if (users && users.length === userIds.length) {
-                await database.sequelize.transaction(async (t) => {
-                    await group.addUsers(userIds, { transaction: t });
-                });
-                return await group.getUsers({ through: { group_id: groupId } });
-            }
-            const userIdsFromDB = users.map((u) => (u.id));
-            const notFoundUserIds = userIds.filter((id) => !userIdsFromDB.includes(id));
-            return { message : `users with ids: ${JSON.stringify(notFoundUserIds)} aren't found` };
+    const group = await database.Group.findOne({ where: { id : groupId } });
+    if (group) {
+        const users = await database.User.findAll({ where: { id : userIds, isDeleted : false } });
+        if (users && users.length === userIds.length) {
+            await database.sequelize.transaction(async (t) => {
+                await group.addUsers(userIds, { transaction: t });
+            });
+            return await group.getUsers({ through: { group_id: groupId } });
         }
-        return { message : `group with id: ${groupId} isn't found` };
-    } catch (error) {
-        throw error;
+        const userIdsFromDB = users.map((u) => (u.id));
+        const notFoundUserIds = userIds.filter((id) => !userIdsFromDB.includes(id));
+        return { status: 'Not Found', message : `users with ids: ${JSON.stringify(notFoundUserIds)} aren't found` };
     }
+    return { status: 'Not Found', message : `group with id: ${groupId} isn't found` };
```

Comment:
You constructing error response inside on repository. In this case your internal/low-level components aware about external/upper components. In this case your architecture became tightly coupled and in is harder to extend.

From my point it is better to throw exception here with some message/reason or return null and only in controller you will construct error message based on controller logic